service/static/js/rest_api.js

```diff
@@ -356,4 +357,30 @@ $(function () {
         })
     })
 
+    // Delete an item from the list
+    $("#wishlist-items-table").on("click", ".item-delete-btn", function(evnt){
+        // make a call to the endpoint 
+        const btn = $(evnt.target)
+        const ids = btn.data("wishlist-and-item-id")
+        const tokens = ids.split(":")
+
+        $.ajax({
+            method: "DELETE",
+            url: `/wishlists/${tokens[0]}/items/${tokens[1]}`,
+            contentType: "application/json",
+            data: '',
+        }).done(function(res, statusText, jqXHR){
+            // update visible list w/o changing success flash
+            if (jqXHR.status === 204) {
+                // Success - update visible list without changing success flash
+                retrieveAndUpdateWishlist(tokens[0], false);
```

Comment:
I think for the retrieveAndUpdateWishlist we probably shouldn't deal with the flash message at all (except on retrieval failure).  One thing to be cautious of when creating a function that has asyncronous internal calls (i.e. $.ajax() ).  You can have some weird bugs arise.  Like it's possible that the flash_message in line 377 might fire before the ajax calls have completed inside of retrieveAndUpdateWishlist.  I think for our purposes this is fine, but just raising it as food for thought.

Reply:
Gotcha, I was considering this as a chain. I'm not too familiar with JS, so I'll look into how I can fix that.